express-notes-api/index.js

```diff
@@ -0,0 +1,90 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const fs = require('fs');
+
+app.use(express.json());
+
+const newJSON = JSON.stringify(data, null, 2);
+
+app.get('/api/notes', (req, res) => {
+  const array = [];
+  if(!data.notes){
+    return [];
+  }
+  for (const index in data.notes) {
+    array.push(data.notes[index]);
+  }
+   res.status(200).json(array);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if (isNaN(req.params.id)){
+    res.status(400).send(`Error: Id must be a positive integer`);
+  } else if (data.notes[req.params.id] ===undefined){
+    res.status(404).send(`Error: Cannot find note with id ${req.params.id} `);
+  } else{
+    res.status(200).json(data.notes[req.params.id]);
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  if(!req.body.content){
+    res.status(400).send(`Error: Content is a required field`);
+  } else{
+    data.notes[data.nextId] = req.body;
+    data.notes[data.nextId].id = data.nextId;
+    res.status(201).send(data.notes[data.nextId]);
+    data.nextId++;
+    fs.writeFile('data.json', newJSON, 'utf8', err=>{
```

Comment:
you can't use `newJSON` here since it was created before the update. After you update the data, then you want to convert that to JSON and write it to a file